framework/include/constraints/AutomaticMortarGeneration.h

```diff
@@ -178,6 +177,26 @@ class AutomaticMortarGeneration : public ConsoleStreamInterface
    */
   std::vector<Point> getNodalNormals(const Elem & secondary_elem) const;
 
+  /**
+   * @return The nodal tangents associated with the provided \p secondary_elem
+   */
+  std::array<std::vector<Point>, 2> getNodalTangents(const Elem & secondary_elem) const;
+
+  /**
+   * @return The mapping from secondary interior parent nodes to lower dimensional nodes
+   */
+  std::map<unsigned int, unsigned int>
+  getSecondaryIpToLowerElementMap(const Elem & _lower_secondary_elem) const;
+
+  /**
+   * @return The mapping from primary interior parent nodes to its corresponding lower dimensional
+   * nodes
+   */
+  std::map<unsigned int, unsigned int>
+  getPrimaryIpToLowerElementMap(const Elem & _primary_elem,
+                                const Elem & _primary_elem_ip,
+                                const Elem & _lower_secondary_elem) const;
```

Comment:
For these methods I might mention in the doxygen string that these maps are built on-the-fly because every time I see them I wonder why we are returning by value and not by reference